src/test/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriterFlushTests.java

```diff
@@ -264,16 +271,21 @@ public void testFlush_WithQuantization() {
 
             IntStream.range(0, vectorsPerField.size()).forEach(i -> {
                 try {
-                    verify(knn990QuantWriterMockedConstruction.constructed().get(0)).writeState(i, quantizationState);
-                    verify(nativeIndexWriter).flushIndex(expectedVectorValues.get(i), vectorsPerField.get(i).size());
+                    if (vectorsPerField.get(i).isEmpty()) {
+                        verify(knn990QuantWriterMockedConstruction.constructed().get(0), never()).writeState(i, quantizationState);
```

Comment:
We don't want any interactions with quantwriter in this case right?
```suggestion
                        verifyNoInteractions(knn990QuantWriterMockedConstruction);
```